Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Components/Pages/PagesControllerImpl.cs

```diff
@@ -1,1485 +1,1507 @@
+                        if (term == null)
+                        {
+                            var termId = termController.AddTerm(new Term(tag, string.Empty, vocabularyId));
+                            term = termController.GetTerm(termId);
+                        }
+
+                        tab.Terms.Add(term);
+                    }
+                }
+            }
+
+            // icons
+            if (pageSettings.IconFile != null && pageSettings.IconFile.fileId > 0)
+            {
+                tab.IconFile = FileManager.Instance.GetFile(pageSettings.IconFile.fileId).RelativePath;
+            }
+            else
+            {
+                tab.IconFile = null;
+            }
+            if (pageSettings.IconFileLarge != null && pageSettings.IconFileLarge.fileId > 0)
+            {
+                tab.IconFileLarge = FileManager.Instance.GetFile(pageSettings.IconFileLarge.fileId).RelativePath;
+            }
+            else
+            {
+                tab.IconFileLarge = null;
+            }
+
+        }
+
+        protected IOrderedEnumerable<KeyValuePair<int, string>> GetLocales(int portalId)
+        {
+            var locales = new Lazy<Dictionary<string, Locale>>(() => LocaleController.Instance.GetLocales(portalId));
+            return locales.Value.Values.Select(local => new KeyValuePair<int, string>(local.KeyID, local.EnglishName)).OrderBy(x => x.Value);
+        }
+
+        protected IEnumerable<KeyValuePair<int, string>> GetSiteAliases(int portalId)
+        {
+            var aliases = PortalAliasController.Instance.GetPortalAliasesByPortalId(portalId);
+            return aliases.Select(alias => new KeyValuePair<int, string>(alias.KeyID, alias.HTTPAlias)).OrderBy(x => x.Value);
+        }
+
+        protected int? GetPrimaryAliasId(int portalId, string cultureCode)
+        {
+            var aliases = PortalAliasController.Instance.GetPortalAliasesByPortalId(portalId);
+            var primary = aliases.Where(a => a.IsPrimary
+                                             && (a.CultureCode == cultureCode || String.IsNullOrEmpty(a.CultureCode)))
+                .OrderByDescending(a => a.CultureCode)
+                .FirstOrDefault();
+            return primary == null ? (int?)null : primary.KeyID;
+        }
+
+        private static string GetExternalUrlRedirection(string url)
+        {
+            if (url == null)
+            {
+                return null;
+            }
+
+            if (url.ToLower() == "http://")
+            {
+                return string.Empty;
+            }
+
+            if (url.StartsWith("//"))
+            {
+                return url;
+
+            }
+
+            if (url.IndexOf("://") != -1)
+
+            {
+                return Globals.AddHTTP(url);
+            }
+
+            if (!url.StartsWith("/"))
+            {
+                return "/" + url;
+            }

```

Comment:
```suggestion
```

And one more thing that comes to mind is we should not automatically add the initial `/` because it would break other types of valid urls such as `tel:+1234567890`

Reply:
Great point - testing now.